internal/repositories/devicedefinition/devicedefinition.go

```diff
@@ -0,0 +1,224 @@
+	if err != nil {
+		return nil, err
+	}
+
+	if filterBy.Manufacturer == nil || *filterBy.Manufacturer == "" {
+		return nil, gqlerror.Errorf("Provide exactly one `manufacturer`.")
+	}
+
+	var conditions []string
+	if filterBy.Year != nil && (*filterBy.Year > 1980 && *filterBy.Year < 2999) {
+		conditions = append(conditions, fmt.Sprintf("year = %d", *filterBy.Year))
+	}
+	if filterBy.Model != nil && len(*filterBy.Model) > 0 {
+		conditions = append(conditions, fmt.Sprintf("model = '%s'", *filterBy.Model))
+	}
+	if filterBy.ID != nil && len(*filterBy.ID) > 0 {
+		conditions = append(conditions, fmt.Sprintf("id = '%s'", *filterBy.ID))
+	}
+
+	whereClause := strings.Join(conditions, " AND ")
+	if whereClause != "" {
+		whereClause = " WHERE " + whereClause
+	}
+
+	tableName := "_80001_8439"
+
+	countParams := map[string]string{
+		"statement": fmt.Sprintf("SELECT count(*) FROM %s%s", tableName, whereClause),
+	}
+	var modelCountTableland []DeviceDefinitionTablelandCountModel
+	if err = r.QueryTableland(countParams, &modelCountTableland); err != nil {
+		return nil, err
+	}
+
+	totalCount := modelCountTableland[0].Count
+
+	queryParams := map[string]string{
+		"statement": fmt.Sprintf("SELECT * FROM %s%s LIMIT %d OFFSET %d", tableName, whereClause, limit, 1),
+	}
+
+	var modelTableland []DeviceDefinitionTablelandModel
+	if err = r.QueryTableland(queryParams, &modelTableland); err != nil {
+		return nil, err
+	}
+
+	var all []*gmodel.DeviceDefinition
+	for _, item := range modelTableland {
+		all = append(all, ToAPI(&item))
+	}
+
+	// We assume that cursors come from real elements.
+	hasNext := before != nil
+	hasPrevious := after != nil
+
+	if first != nil && len(all) == limit+1 {
+		hasNext = true
+		all = all[:limit]
+	} else if last != nil && len(all) == limit+1 {
+		hasPrevious = true
+		all = all[:limit]
+	}
+
+	if last != nil {
+		slices.Reverse(all)
+	}
+
+	var errList gqlerror.List
+	var endCur, startCur *string
+	if len(all) != 0 {
+		//ec := helpers.IDToCursor(all[len(all)-1].ID)
+		//endCur = &ec
+		//
+		//sc := helpers.IDToCursor(all[0].ID)
+		//startCur = &sc
+	}
+
+	edges := make([]*gmodel.DeviceDefinitionEdge, len(all))
+	nodes := make([]*gmodel.DeviceDefinition, len(all))
+
+	for i, dv := range all {
+
+		edges[i] = &gmodel.DeviceDefinitionEdge{
+			Node: dv,
+			//Cursor: helpers.IDToCursor(dv.ID),
+		}
+
+		nodes[i] = dv
+	}
+
+	res := &gmodel.DeviceDefinitionConnection{
+		Edges: edges,
+		Nodes: nodes,
+		PageInfo: &gmodel.PageInfo{
+			EndCursor:       endCur,
+			HasNextPage:     hasNext,
+			HasPreviousPage: hasPrevious,
+			StartCursor:     startCur,
+		},
+		TotalCount: int(totalCount),
+	}
+	if errList != nil {
+		return res, errList
+	}
+	return res, nil
+}
+
+func (r *DeviceDefinitionRepository) QueryTableland(queryParams map[string]string, result interface{}) error {
+	fullURL, err := url.Parse(r.Settings.TablelandAPIGateway)
+	if err != nil {
+		return err
+	}
+
+	fullURL.Path = path.Join(fullURL.Path, "api/v1/query")
```

Comment:
Use this Join Path function it is safer https://pkg.go.dev/net/url#URL.JoinPath
```suggestion
	fullURL = fullURL.JoinPath(fullURL.Path, "api/v1/query")
```